libraries/libfc/src/crypto/bls_private_key.cpp

```diff
@@ -25,38 +25,49 @@ namespace fc::crypto::blslib {
       return bls_private_key(v);
    }
 
-   template<typename Data>
-   Data from_wif( const string& wif_key )
+   string to_wif(vector<uint8_t> secret, const fc::yield_function_t& yield )
    {
-   /*   auto wif_bytes = from_base58(wif_key);
+      FC_ASSERT(secret.size() == 32);
+
+      std::array<uint8_t, 32> v2;
+
+      std::copy(secret.begin(), secret.end(), v2.begin());
+
+      const size_t size_of_data_to_hash = 32 + 1;
+      const size_t size_of_hash_bytes = 4;
+      char data[size_of_data_to_hash + size_of_hash_bytes];
+      data[0] = (char)0x80; // this is the Bitcoin MainNet code
+      memcpy(&data[1], (const char*)&v2, 32);
+      sha256 digest = sha256::hash(data, size_of_data_to_hash);
+      digest = sha256::hash(digest);
+      memcpy(data + size_of_data_to_hash, (char*)&digest, size_of_hash_bytes);
+
+      return to_base58(data, sizeof(data), yield);
+   }
+
+   std::vector<uint8_t> from_wif( const string& wif_key )
+   {
+      auto wif_bytes = from_base58(wif_key);
       FC_ASSERT(wif_bytes.size() >= 5);
+
       auto key_bytes = vector<char>(wif_bytes.begin() + 1, wif_bytes.end() - 4);
       fc::sha256 check = fc::sha256::hash(wif_bytes.data(), wif_bytes.size() - 4);
       fc::sha256 check2 = fc::sha256::hash(check);
 
       FC_ASSERT(memcmp( (char*)&check, wif_bytes.data() + wif_bytes.size() - 4, 4 ) == 0 ||
                 memcmp( (char*)&check2, wif_bytes.data() + wif_bytes.size() - 4, 4 ) == 0 );
 
-      return Data(fc::variant(key_bytes).as<typename Data::data_type>());*/
+      std::vector<uint8_t> v2(32);
+      std::copy(key_bytes.begin(), key_bytes.end(), v2.begin());

```

Comment:
Nothing prevents `key_bytes` from being larger than `32` which would cause a memory corruption here.

Reply:
I think you could avoid this copy by using a `std::vector<uint8_t>` internally and casting to `char*`.